libs/constellation/src/constellation.cpp

```diff
@@ -691,50 +739,225 @@ bool Constellation::Run(UriSet const &initial_peers, core::WeakRunnable bootstra
       // client connections. By delaying these notify() calls to the point when the node believes
       // it has successfully synchronised this ensures that cleaner network start up
       //
-      reactor_.Attach(std::move(bootstrap_monitor));
+      reactor_.Attach(bootstrap_monitor);
       start_up_in_progress = false;
 
       FETCH_LOG_INFO(LOGGING_NAME, "Startup complete");
     }
 
-    // update the uptime counter
+    // update the up time counter
     uptime_->increment();
   }
 
-  //---------------------------------------------------------------
-  // Step 3. Tear down
-  //---------------------------------------------------------------
+  return true;
+}
 
-  FETCH_LOG_INFO(LOGGING_NAME, "Shutting down...");
+void Constellation::OnTearDownExternalNetwork()
+{
+  FETCH_LOG_INFO(LOGGING_NAME, "OnTearDownExternalNetwork()");
+
+  if (http_)
+  {
+    http_->Stop();
+    ResetItem(http_);
+  }
+
+  if (tx_processor_)
+  {
+    tx_processor_->Stop();
+  }
+
+  if (execution_manager_)
+  {
+    execution_manager_->Stop();
+  }
 
-  http_.Stop();
-  tx_processor_.Stop();
   reactor_.Stop();
+
   if (agent_network_)
   {
     agent_network_->Stop();
   }
 
-  execution_manager_->Stop();
-  storage_.reset();
-  lane_services_.Stop();
-  muddle_->Stop();
-  internal_muddle_->Stop();
-  http_network_manager_.Stop();
-  network_manager_.Stop();
-  http_open_api_module_->Reset(nullptr);
+  if (beacon_network_)
+  {
+    beacon_network_->Stop();
+  }
 
-  FETCH_LOG_INFO(LOGGING_NAME, "Shutting down...complete");
+  if (muddle_)
+  {
+    muddle_->Stop();
+  }
 
-  return startup_success;
+  lane_services_.StopExternal();
+
+  ResetItem(synergetic_miner_);
+  ResetItem(dag_service_);
+
+  http_modules_.clear();
+  ResetItem(health_check_module_);
+  ResetItem(http_open_api_module_);
+
+  ResetItem(messenger_api_);
```

Comment:
Won't this happen anyway? Is this so you can do it in a particular order, or for easier debugging?

Reply:
Yes this is over kill but I want to have the shutdown exactly when when this function is called